-Wincompatible-pointer-types error in init.c, if gcc version >= 14

When remaking my lwip-using unikernel on a new computer the build terminated on this error:

```
/home/akira/avar_src/app-mongod403/w/libs/lwip/init.c: In function ‘liblwip_init’:
/home/akira/avar_src/app-mongod403/w/libs/lwip/init.c:402:58: error: passing argument 2 of ‘dns_setserver’ from incompatible pointer type [-Wincompatible-pointer-types]
  402 |                                 dns_setserver(nb_dns4++, &dns4);
      |                                                          ^~~~~
      |                                                          |
      |                                                          ip4_addr_t * {aka struct ip4_addr *}
```

It appears that gcc version >= 14 will make `-Wincompatible-pointer-types` an error instead of warning by default: https://gcc.gnu.org/gcc-14/porting_to.html

My workaround for now is simply to add `-Wno-incompatible-pointer-types` (or `-Wno-error=incompatible-pointer-types`?)  to the _LIBLWIP_CFLAGS_ line in lwip's `Makefile.uk`.

--- a/init.c
+++ b/init.c
@@ -165,7 +165,7 @@ static int liblwip_init(struct uk_init_ctx *ictx __unused)
 	ip4_addr_t *gw4_arg;
 	const char *hostname_arg;
 #if LWIP_DNS
-	ip4_addr_t dns4;
+	ip_addr_t dns;
 	unsigned int nb_dns4 = 0;
 #endif /* LWIP_DNS */
 #endif /* LWIP_IPV4 */
@@ -393,13 +393,18 @@ no_conf:
 		if (nb_dns4 < DNS_MAX_SERVERS) {
 			strcfg = uk_netdev_einfo_get(dev, UK_NETDEV_IPV4_DNS0);
 			if (strcfg) {
-				if (ip4addr_aton(strcfg, &dns4) != 1) {
+#if LWIP_IPV6
+				dns.type = IPADDR_TYPE_V4;
+				if (ip4addr_aton(strcfg, &dns.u_addr.ip4) != 1) {
+#else /* LWIP_IPV6 */
+				if (ip4addr_aton(strcfg, &dns) != 1) {
+#endif /* LWIP_IPV6 */
 					uk_pr_err("Failed to parse DNS server address: %s\n",
 						  strcfg);
 					goto dns_secondary;
 				}
 
-				dns_setserver(nb_dns4++, &dns4);
+				dns_setserver(nb_dns4++, &dns);
 				uk_pr_info("%c%c%u: Primary DNS server: %s\n",
 					   nf->name[0], nf->name[1], nf->num,
 					   strcfg);
@@ -411,13 +416,18 @@ dns_secondary:
 		if (nb_dns4 < DNS_MAX_SERVERS) {
 			strcfg = uk_netdev_einfo_get(dev, UK_NETDEV_IPV4_DNS1);
 			if (strcfg) {
-				if (ip4addr_aton(strcfg, &dns4) != 1) {
+#if LWIP_IPV6
+				dns.type = IPADDR_TYPE_V4;
+				if (ip4addr_aton(strcfg, &dns.u_addr.ip4) != 1) {
+#else /* LWIP_IPV6 */
+				if (ip4addr_aton(strcfg, &dns) != 1) {
+#endif /* LWIP_IPV6 */
 					uk_pr_err("Failed to parse DNS server address: %s\n",
 						  strcfg);
 					goto dns_done;
 				}
 
-				dns_setserver(nb_dns4++, &dns4);
+				dns_setserver(nb_dns4++, &dns);
 				uk_pr_info("%c%c%u: Secondary DNS server: %s\n",
 					   nf->name[0], nf->name[1], nf->num,
 					   strcfg);

